csrc/device_lower/pass/index.cpp

```diff
@@ -2578,6 +2578,41 @@ Val* getOuterStrideBytes(TensorView* tv, const MmaOp* mma) {
   return SimplifyingIrBuilder::mulExpr(stride, dataTypeSize(tv->dtype()));
 }
 
+namespace {
+
+Val* indexBlackwellMmaOutput(
+    const MmaOp* mma,
+    const std::vector<ForLoop*>& for_loops) {
+  TensorView* tmem_tv = mma->out()->as<TensorView>();
+  NVF_ERROR(tmem_tv->getMemoryType() == MemoryType::Tensor, "Invalid tmem_tv");
+  const auto& tmem_info = GpuLower::current()->tmemInfo();
+  const auto& tensor_indexer = GpuLower::current()->tensorIndexer();
+
+  const std::vector<IterDomain*>& column_allocation_domain =
+      tmem_info.allocation.getTVInfo(tmem_tv).column_allocation;
+
+  std::vector<Val*> indices = tensor_indexer.getIndexFor(
+      mma, true, column_allocation_domain, for_loops);
+  Val* stride = tmem_tv->fusion()->oneVal();
+  Val* column_index = tmem_tv->fusion()->zeroVal();
+  for (const auto& [id, idx] :
+       zip(std::ranges::views::reverse(column_allocation_domain),
+           std::ranges::views::reverse(indices))) {
```

Comment:
Just curious, does this work?
```suggestion
  for (const auto& [id, idx] :
       std::ranges::views::reverse(zip(column_allocation_domain,indices))) {
```

Reply:
It works. There used to be a bug in zip making this not working, but that bug has been fixed.